shims/write/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources
+
+import org.apache.spark.{SparkContext, TaskContext}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.SQLExecution
+import org.apache.spark.sql.execution.datasources.BasicWriteJobStatsTracker._
+import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics}
+import org.apache.spark.util.SerializableConfiguration
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+
```

Comment:
Is this file override spark class file? Why add new module `write`, you can add it to module `common` if it is same for all the spark version.

Can we have other solution to fix this bug?

Reply:
Thanks for your review. We add a write directory here to solve the dependency problem. Because we need to access `FakeRow `under the `gluten-core` package in the `common ` directory, but since `gluten-core` depends on `commom`, a circular dependency problem will be formed. And This fix is to replace the `BasicWriteTaskStat ` class in vanilla spark. I think it is more appropriate to put it in Shim layer. Or do you have any better suggestion?

Reply:
If it only use `FakeRow`, I think we can also put FakeRow to module `common`.

Other projects dependency maybe
-spark33
   -gluten-core
   -gluten-data

This will solve the problem, if this problems occur too many times, we can try to refactor code.

Reply:
`FakeRow `is defined in the Gluten project and is not suitable for shim layer. How about adding a new `gluten-datasource` package and move all the classes related to native write to this package?

Reply:
@binmahone @rui-mo Do you have any suggestion?

Reply:
Offline discussed with @binmahone , prefer to move `FakeRow ` into `common`. @jinchengchenghh already updated, please help to review again. Thanks.